File: app/src/main/java/cu/axel/smartdock/models/DockApp.java
package cu.axel.smartdock.models;

import java.util.ArrayList;

import android.graphics.drawable.Drawable;

public class DockApp extends App {
    private final ArrayList<AppTask> tasks;

    public DockApp(String name, String packageName, Drawable icon) {
        super(name, packageName, icon);

        tasks = new ArrayList<>();
    }

    public DockApp(AppTask task) {
        super(task.getName(), task.getPackageName(), task.getIcon());
        tasks = new ArrayList<>();
        tasks.add(task);
    }

    public void addTask(AppTask task) {
        tasks.add(task);
    }

    public ArrayList<AppTask> getTasks() {
        return tasks;
    }

    @Override
    public Drawable getIcon() {
        return tasks.size() == 1 ? tasks.get(0).getIcon() : super.getIcon();
    }
}

File: app/src/main/java/cu/axel/smartdock/models/DockApp.kt
package cu.axel.smartdock.models

import android.graphics.drawable.Drawable

class DockApp : App {
    val tasks: ArrayList<AppTask>

    constructor(name: String, packageName: String, icon: Drawable) : super(name, packageName, icon) {
        tasks = ArrayList()
    }

    constructor(task: AppTask) : super(task.name, task.packageName, task.icon) {
        tasks = ArrayList()
        tasks.add(task)
    }

    fun addTask(task: AppTask) {
        tasks.add(task)
    }

    override val icon: Drawable
        get() = if (tasks.size == 1) tasks[0].icon else super.icon
}
